Example/Example/Shared/Configuration.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Configuration.swift
+//  Example
+//
+//  Created by Ryan Davis on 4/8/20.
+//  Copyright © 2020 Afterpay. All rights reserved.
+//
+
+import Foundation
+import os.log
+
+private extension TimeInterval {
+  static var oneDay: Self { 60 * 60 * 24 }
+}
+
+private enum UserDefaultsKey {
+  static var configuration: String { #function }
+  static var lastFetchDate: String { #function }
+}
+
+func getCachedConfiguration(
+  userDefaults: UserDefaults = .standard,
+  now: @escaping () -> Date = Date.init,
+  completion: @escaping (Result<ConfigurationResponse, Error>) -> Void
+) {
+  if
+    let configuration = userDefaults.data(forKey: UserDefaultsKey.configuration),
+    let lastFetchDate = userDefaults.object(forKey: UserDefaultsKey.lastFetchDate) as? Date,
+    lastFetchDate.addingTimeInterval(.oneDay) > now()
+  {
+    do {
+      let configuration = try JSONDecoder().decode(ConfigurationResponse.self, from: configuration)
```

Comment:
I can see some use for re-factoring where you can retrieve the data from either the userDefaults *or* the network and decode in one play